lib/irb.rb

```diff
@@ -1513,8 +1513,8 @@ class Binding
   #
   # See IRB for more information.
   def irb(show_code: true)
-    # Setup IRB with the current file's path and no command line arguments
-    IRB.setup(source_location[0], argv: [])
+    # Setup IRB with the current file's path and current session command line arguments
+    IRB.setup(source_location[0], argv: IRB.conf[:ARGV])
```

Comment:
Don't really know how to test this, other than manually steps:

```sh
$ mkdir -p tmp
$ echo "puts 'irbrc file loaded'" > tmp/.irbrc 
$ bundle exec rake install
$ % cd tmp && echo "binding.irb" | irb -f && cd ..
```

Before:

```
Switch to inspect mode.
binding.irb
irbrc file loaded
Switch to inspect mode.

nil
```

After:

```
Switch to inspect mode.
binding.irb
Switch to inspect mode.

nil
```


---

I thought about doing this test:

```rb
    def test_new_session_inherits_parent_config
      write_rc <<~RUBY
        puts 'irbrc file loaded'
      RUBY

      write_ruby <<~'RUBY'
        require 'irb'
        IRB.start
      RUBY

      output = run_ruby_file(["-f"]) do
        type "binding.irb"
        type "exit!"
        type "exit!"
      end

      refute output.include?("irbrc file loaded")
    end
```

And change the helper method `ruby_ruby_file` to:
```rb
    def run_ruby_file(cmd_args=[], &block)
      cmd = [EnvUtil.rubybin, "-I", LIB, @ruby_file.to_path] + cmd_args
```

But the test times-out. I think it's because `safe_gets` or writing to the command in `PTY.spawn` is not working this way :/